morphir/runtime/src/org/finos/morphir/runtime/quick/EvaluatorQuick.scala

```diff
@@ -132,7 +142,7 @@ object EvaluatorQuick {
       case TT.Reference(_, typeName, typeArgs) =>
         val lookedUp    = dists.lookupTypeSpecification(typeName.packagePath, typeName.modulePath, typeName.localName)
         val conceptArgs = typeArgs.map(typeToConcept(_, dists, boundTypes))
-        lookedUp.getOrElse(throw new Exception(s"Could not find spec for $typeName")) match {
+        lookedUp.getOrElse(throw new UnexpectedType(s"Could not find spec for $typeName")) match {
```

Comment:
There's a particular error for something being missing, and the return from Dists should be an either with a helpful error type for specifying what's missing